src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -436,29 +460,17 @@ void DnsMetricsBucket::process_dnstap(bool deep, const dnstap::Dnstap &payload)
 {
     std::unique_lock lock(_mutex);
 
+    pcpp::ProtocolType l3;
     if (payload.message().has_socket_family()) {
         if (payload.message().socket_family() == dnstap::INET6) {
-            ++_counters.IPv6;
+            l3 = pcpp::IPv6;
```

Comment:
actually we have two process_dns_layer. but your affirmation was right in case of deep sampling and no query/response message. I've added an else there now to process  this correctly